src/panels/logbook/ha-logbook.ts

```diff
@@ -303,17 +303,22 @@ export class HaLogbook extends LitElement {
         )
       : this._logbookEntries;
 
-  private _processNewEntries = (
-    newEntries: LogbookEntry[],
+  private _processStreamMessage = (
+    streamMessage: LogbookStreamMessage,
     purgeBeforePythonTime: number | undefined
   ) => {
     // Put newest ones on top. Reverse works in-place so
     // make a copy first.
-    newEntries = [...newEntries].reverse();
+    const newEntries = [...streamMessage.events].reverse();
     if (!this._logbookEntries) {
       this._logbookEntries = newEntries;
       return;
     }
+    if (!newEntries.length) {
```

Comment:
This will fix the `Cannot read properties of undefined (reading 'when')` error logbook data is empty for the live period.

I noticed it when I put the prerelease from an hour ago on my RPi3 (not powerful enough to run the dev setup for frontend so I have to wait for releases to test)

Reply:
Besides that logbook is quite responsive on my RPi3 test machine with the crappiest micro sd card I have that I could get to boot for testing.